File: ClientApp/RetroArcadeMachines/src/app/feature-modules/locations/models/location-details.model.ts
import { GridConfig } from '@core/modules/grid/models/grid-config.model';
import { LocationOverview } from '@shared/models/location-overview.model';
import { GameOverview } from 'src/app/shared/models/game-overview.model';
import { Address } from './address.model';
import { BusinessHours } from './business-hours.model';

export class LocationDetails extends LocationOverview {
    synopsis: string;
    emailAddress: string;
    website: string;
    lat: string;
    lng: string;
    address: Address;
    phoneNumberList: string[];
    businessHoursList: BusinessHours[];
    gameOverviewList: GameOverview[];
    imageUrlList: string[];
    gameCollectionTable: GridConfig<GameOverview>;
}

File: Src/RetroArcadeMachines.Shared.Models/LocationDetailsModel.cs
using Amazon.DynamoDBv2.DataModel;
using System;
using System.Collections.Generic;

namespace RetroArcadeMachines.Shared.Models
{
    [DynamoDBTable("LocationsDetails")]
    public class LocationDetailsModel : LocationBaseModel
    {
        public string Synopsis { get; set; }
        public bool IsRetroGamesOnly { get; set; }
        public string EmailAddress { get; set; }
        public string Website { get; set; }
        public string Lat { get; set; }
        public string Lng { get; set; }
        public AddressModel Address { get; set; }
        public List<PhoneNumberModel> PhoneNumberList { get; set; }
        public List<BusinessHoursModel> BusinessHoursList { get; set; }
        public Dictionary<string, string> GameOverviewList { get; set; }
        public List<string> ImageUrlList { get; set; }
    }
}
